cpu/esp8266/include/periph_cpu.h

```diff
@@ -264,6 +265,19 @@ typedef enum {
     HSPI = 1,         /**< HSPI interface controller */
 } spi_ctrl_t;
 
+/**
+ * @brief   Override SPI clock speed values
+ * @{
+ */
+#define HAVE_SPI_CLK_T
+typedef enum {
+    SPI_CLK_100KHZ = KHZ(100), /**< drive the SPI bus with 100KHz */
+    SPI_CLK_400KHZ = KHZ(400), /**< drive the SPI bus with 400KHz */
+    SPI_CLK_1MHZ   = MHZ(1),   /**< drive the SPI bus with 1MHz */
+    SPI_CLK_5MHZ   = MHZ(5),   /**< drive the SPI bus with 5MHz */
+    SPI_CLK_10MHZ  = MHZ(10),  /**< drive the SPI bus with 10MHz */
+} spi_clk_t;
+
```

Comment:
Why was this change necessary if the enum values are (no longer?) used?

Reply:
The enum values only go from 0 to 4, with this you can just write the desired SPI frequency to the `clk` parameter instead.

Reply:
Where is the original enum definition? I somehow doubt it is a good idea to override those only for a single platform?

Reply:
The previous definition was [here](https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/periph/spi.h#L184) - you will find that overwriting this is done by most platforms already since the original interface is pretty bad/limiting.

Reply:
Hum, maybe worth an issue to clean up the original interface then in a subsequent PR?

Reply:
We might introduce a new feature for platforms where you can select arbitrary SPI clocks, so drivers can rely on this.